src/CMakeLists.txt

```diff
@@ -294,15 +289,39 @@ ELSE()
 	MESSAGE(FATAL_ERROR "Asked for unknown SHA1 backend ${SHA1_BACKEND}")
 ENDIF()
 
-# Include PCRE and its POSIX regex compatibility layer when it is required
-IF (HAVE_REGCOMP_L)
+# Specify regular expression implementation
+IF(REGEX STREQUAL "")
+	CHECK_SYMBOL_EXISTS(regcomp_l "regex.h;xlocale.h" HAVE_REGCOMP_L)
+	CHECK_SYMBOL_EXISTS(pcre_regcomp "pcreposix.h" HAVE_PCRE)
+
+	IF (HAVE_REGCOMP_L)
+		SET(REGEX "regcomp_l")
+	ELSEIF (HAVE_PCRE)
+		SET(REGEX "pcre")
+	ELSE ()
+		SET(REGEX "builtin")
+	ENDIF ()
+ENDIF()
+
+IF(REGEX STREQUAL "regcomp_l")
 	ADD_FEATURE_INFO(regex ON "using system regcomp_l")
-ELSE()
+	SET(GIT_REGEX_REGCOMP_L 1)
+ELSEIF(REGEX STREQUAL "pcre")
+	ADD_FEATURE_INFO(regex ON "using system PCRE")
+	SET(GIT_REGEX_PCRE 1)
```

Comment:
Shouldn't we detect here whether system-pcre exists and set up includes/libraries? Or did I miss a past commit that already sets this up for us?

Reply:
I added this - we will now look for PCRE and, if it's installed, prefer it to our own PCRE routines.  

A user can also opt-in to using PCRE2 if it is installed, but we do _not_ autodetect PCRE2.  The POSIX compatibility layer for PCRE2 is broken on Debian-based installs (the library symbols are prefixed with `PCRE2`, eg, `PCRE2regcomp`, but the headers make no mention of this.  They were supposed to set up a layer of indirection, eg `#define regcomp PCRE2regcomp` but this was neglected.  This is being fixed but we cannot rely on it yet.)